### PR TITLE
Separate CXX flags for MSVC and GCC

### LINKED ISSUE #42: Building DEBUG version enables -Wall which causes external library header files to produce  large amounts of warnings
There should be some way to disable warnings on external library header files.

### PR DESCRIPTION
**Pull request by MonJamp**
Needs testing on Windows
Fixes #42

Edit: Whoops I kinda messed up the PR.

### CODE REVIEW

**Comment by Dyrand** [comment]
This PR depends on other commits include other commits which are within https://github.com/JethroDevon/revengeMusic/pull/48
Possibly work on separate branches for each related changes you might want to make?

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,11 +14,15 @@ include_directories("${CMAKE_SOURCE_DIR}/include/")
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -O0 -Wall")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2")
-IF(NOT WIN32)
+IF(MSVC)
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -W4")
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
```
Comment (Dyrand) [inline CMakeLists.txt]: Equivalent flag for optimization in MSVC is probably /O2

File: CMakeLists.txt [hunk 2]
```diff
@@ -14,11 +14,15 @@ include_directories("${CMAKE_SOURCE_DIR}/include/")
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -O0 -Wall")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2")
-IF(NOT WIN32)
+IF(MSVC)
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -W4")
```
Comment (Dyrand) [inline CMakeLists.txt]: Is that syntax accepted in MSVC? 
I'm only aware of the flags being accepted with a forward slash: /W4

**Review by Dyrand — CHANGES_REQUESTED** [review]

**Comment by Dyrand** [comment]
I created a merge commit instead of a rebase for the other PR you made that fixed #44, so I reverted it.
This PR fixes both #42 and #44, so it should be alright?

File: CMakeLists.txt (hunk 2 above)
Comment (MonJamp) [reply CMakeLists.txt]: Oddly enough `-Wall` and `-W4` do work in MSVC (maybe due to cmake), but I'll change it to the proper format.

File: CMakeLists.txt [hunk 3]
```diff
@@ -14,11 +14,15 @@ include_directories("${CMAKE_SOURCE_DIR}/include/")
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -O0 -Wall")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2")
-IF(NOT WIN32)
+IF(MSVC)
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /Od /W4")
```
Comment (Dyrand) [inline CMakeLists.txt]: One of these debug flags may be convenient: https://docs.microsoft.com/en-us/cpp/build/reference/z7-zi-zi-debug-information-format

**Review by Dyrand — APPROVED** [review]

**Pull request merged**